internal/provider/openstack/provider_test.go

```diff
@@ -1189,3 +1190,59 @@ func envWithNetworking(net Networking, netCfg string) *Environ {
 		networking: net,
 	}
 }
+
+func (s *providerUnitTests) TestTerminateInstanceNetworkPorts(c *gc.C) {
+	ctrl := gomock.NewController(c)
+	defer ctrl.Finish()
+
+	modelUUID := "test-model"
+	instanceID := instance.Id("test-instance-1")
+
+	testInterfaces := []nova.OSInterface{
+		{PortID: "server-1-port-1"},
+		{PortID: "server-1-port-2"},
+		{PortID: ""},
+	}
+
+	mockNeutron := NewMockNetworkingNeutron(ctrl)
+
+	filter := neutron.NewFilter()
```

Comment:
```suggestion
    // We need to ensure that the query against Openstack ListPortsV2 is passing in a device_id filter
    // so that we know it isn't getting all ports that openstack knows about.
	filter := neutron.NewFilter()
```